### PR TITLE
Enable serverless domains via HTTP

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Adds the ability to go to serverless JSON domains via HTTP and HTTPS.

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
###Testing

Check that going to the installed serverless tutorial still works, i.e., file:///~/serverless/tutorial.json.

Host a copy of tutorial.json on the web and check that you can go to the serverless tutorial via:
- http
- https

Check that the serverless tutorial is included in the goto history (i.e., you can go back to it and go forward to it).

Check that other goto types still work in the Goto dialog, e.g.:
- Going to a position and orientation in the current domain.
- Going to another domain via placename.
- Going to another domain via Web domain name.
- Going to another domain via IP address.
- Going to localhost.
- Going to a person.

**Comment by Darlingnotin** [comment]
Tested everything on the list to test. Everything seems to work. Nice job.

File: libraries/networking/src/AddressManager.cpp [hunk 1]
```diff
@@ -358,11 +358,9 @@ bool AddressManager::handleUrl(const QUrl& lookupUrlIn, LookupTrigger trigger) {
...
 
         return true;
-    } else if (lookupUrl.scheme() == HIFI_URL_SCHEME_FILE) {
-        // TODO -- once Octree::readFromURL no-longer takes over the main event-loop, serverless-domain urls can
-        // be loaded over http(s)
-        // lookupUrl.scheme() == URL_SCHEME_HTTP ||
-        // lookupUrl.scheme() == HIFI_URL_SCHEME_HTTPS ||
+    } else if (lookupUrl.scheme() == HIFI_URL_SCHEME_FILE || lookupUrl.scheme() == HIFI_URL_SCHEME_HTTP
+            || lookupUrl.scheme() == HIFI_URL_SCHEME_HTTPS) {
+
```
Comment (thoys) [inline libraries/networking/src/AddressManager.cpp]: Did you verify that readFromURL no longer takes over the main-event loop? What happens if we have a file that is huge, or a page that times out?

Comment (ctrlaltdavid) [reply libraries/networking/src/AddressManager.cpp]: It still takes over the main event loop, hence the "do not merge", for now.
Am working on making the reading asynchronous, to remove the use of the event loop.

**Comment by ctrlaltdavid** [comment]
Will submit a new PR

**Pull request closed without merging**